hoomd/mpcd/Sorter.cc

```diff
@@ -58,36 +58,25 @@ void mpcd::Sorter::update(uint64_t timestep)
  */
 void mpcd::Sorter::computeOrder(uint64_t timestep)
     {
-    // compute the cell list at current timestep, guarantees owned particles are on rank
     m_cl->compute(timestep);
-
-    ArrayHandle<unsigned int> h_cell_list(m_cl->getCellList(),
-                                          access_location::host,
-                                          access_mode::read);
-    ArrayHandle<unsigned int> h_cell_np(m_cl->getCellSizeArray(),
-                                        access_location::host,
-                                        access_mode::read);
-    const Index2D& cli = m_cl->getCellListIndexer();
-
-    // loop through the cell list to generate the sorting order for MPCD particles
+    m_order.resize(m_mpcd_pdata->getN());
     ArrayHandle<unsigned int> h_order(m_order, access_location::host, access_mode::overwrite);
-    ArrayHandle<unsigned int> h_rorder(m_rorder, access_location::host, access_mode::overwrite);
-    const unsigned int N_mpcd = m_mpcd_pdata->getN();
-    unsigned int cur_p = 0;
-    for (unsigned int idx = 0; idx < m_cl->getNCells(); ++idx)
+
+    // sort indices
+    std::vector<unsigned int> indexes(m_mpcd_pdata->getN());
```

Comment:
You should use `m_order` as the memory for the indexes, rather than a separate vector. Then, you don't need to copy one into the other (you can remove the for loop below).

Reply:
`begin()` and `end()` return iterators, and I think you should be able to use raw pointers instead because they support the same interface. Like:
```c++
ArrayHandle<unsigned int> h_order(m_order, access_location::host, access_mode::overwrite);
std::iota(h_order.data, h_order.data + N, 0);
```
To fill the array with the increasing indexes using `std::iota`.